pulp_rpm/tests/functional/api/test_upload.py

```diff
@@ -47,260 +24,229 @@
-
-    def test_single_request_unit_and_duplicate_unit(self):
-        """Test single request upload unit.
-
-        1. Upload a unit
-        2. Attempt to upload same unit
-        """
-        # Single unit upload
-        upload = self.do_test(self.file_to_use)
-        content = monitor_task(upload.task).created_resources[0]
-        package = self.content_api.read(content)
-        self.assertTrue(package.location_href == RPM_PACKAGE_FILENAME)
-
-        # Duplicate unit
-        upload = self.do_test(self.file_to_use)
-        try:
-            monitor_task(upload.task)
-        except PulpTaskError:
-            pass
-        task_report = self.tasks_api.read(upload.task)
-        msg = "There is already a package with"
-        self.assertTrue(msg in task_report.error["description"])
-
-    def test_upload_non_ascii(self):
-        """Test whether one can upload an RPM with non-ascii metadata."""
-        packages_count = self.content_api.list().count
-        upload = self.do_test(RPM_WITH_NON_ASCII_URL)
-        monitor_task(upload.task)
-        new_packages_count = self.content_api.list().count
-        self.assertTrue((packages_count + 1) == new_packages_count)
 
-    def do_test(self, remote_path):
-        """Upload a Package and return Task of upload."""
-        with NamedTemporaryFile() as file_to_upload:
-            file_to_upload.write(http_get(remote_path))
-            upload_attrs = {"file": file_to_upload.name}
-            return self.content_api.create(**upload_attrs)
+SMALL_CONTENT = SMALL_GROUPS + SMALL_CATEGORY + SMALL_LANGPACK + SMALL_ENVIRONMENTS
+CENTOS8_CONTENT = BIG_GROUPS + BIG_CATEGORY + BIG_LANGPACK + BIG_ENVIRONMENTS
 
 
-class CompsXmlTestCase(PulpTestCase):
-    """
-    Test comps.xml-upload functionality.
+def test_single_request_unit_and_duplicate_unit(
+    delete_orphans_pre, http_get, rpm_package_api, monitor_task, tasks_api_client
+):
+    """Test single request upload unit.
+
+    1. Upload a unit
+    2. Attempt to upload same unit
     """
+    # Single unit upload
+    file_to_use = os.path.join(RPM_UNSIGNED_FIXTURE_URL, RPM_PACKAGE_FILENAME)
+
+    with NamedTemporaryFile() as file_to_upload:
+        file_to_upload.write(http_get(file_to_use))
+        upload_attrs = {"file": file_to_upload.name}
+        upload = rpm_package_api.create(**upload_attrs)
+
+    content = monitor_task(upload.task).created_resources[0]
+    package = rpm_package_api.read(content)
+    assert package.location_href == RPM_PACKAGE_FILENAME
+
+    # Duplicate unit
+    with NamedTemporaryFile() as file_to_upload:
+        file_to_upload.write(http_get(file_to_use))
+        upload_attrs = {"file": file_to_upload.name}
+        upload = rpm_package_api.create(**upload_attrs)
+
+    try:
+        monitor_task(upload.task)
+    except PulpTaskError:
+        pass
+    task_report = tasks_api_client.read(upload.task)
+    msg = "There is already a package with"
+    assert msg in task_report.error["description"]
```

Comment:
NOTE THESE LINES - we reverted the dup-content-upload PR in 3.18; these lines restore this test back to "we see a failure in this context".